e2e/testcases/helm_sync_test.go

```diff
@@ -879,3 +913,148 @@ func helmChartVersion(chartVersion string) nomostest.Sha1Func {
 		return chartVersion, nil
 	}
 }
+
+// fetchServiceAccountKeyFile downloads a service account key from google
+// secret manager, using the specified `gsaKeySecretID`. If the key has expired,
+// a new key is generated and stored as a new version of the secret in secret
+// manager. If successful, the old expired key is deleted. Then the path to the
+// new key file is returned. The key file will be deleted when the current
+// test ends.
+func fetchServiceAccountKeyFile(nt *nomostest.NT, projectID, gsaKeySecretID, gsaEmail, gsaName string) (string, error) {
```

Comment:
Might be worth wrapping this function in a mutex to synchronize threads in a given test execution, in case someone adds another call to this function in the future.